src/frontend/src/editor.tsx

```diff
@@ -4,12 +4,14 @@ import type { CodeFile } from './types'
 import { retrieve, store } from './store.ts'
 import { Tabs, findActive } from './tabs'
 import type { Editor } from './monacoEditor'
+import type { Accessor } from 'solid-js'
 
 interface EditorProps {
   runCode: (files: CodeFile[]) => void
+  running: Accessor<boolean>
 }
 
-export default function ({ runCode }: EditorProps) {
+export default function ({ runCode, running }: EditorProps) {
```

Comment:
I think the correct way to do this is to use 
```suggestion
export default function (props: EditorProps) {
```
Then you can access the props directly like `disabled={props.running}`.